packages/plugin-core/src/commands/VaultAddCommand.ts

```diff
@@ -133,6 +133,83 @@ export class VaultAddCommand extends BasicCommand<CommandOpts, CommandOutput> {
     };
   }
 
+  async gatherVaultSelfContained(
+    sourceType: VaultRemoteSource
+  ): Promise<CommandOpts | undefined> {
+    // If the vault name already exists, creating a vault with the same name would break things
+
+    if (sourceType === "local") {
+      // Local vault
+      // For self contained vaults, we'll have the vault name match the folder for
+      // now. We can make this flexible later if that's a better UX, or give
+      // instructions on the wiki on how to change the name later.
+      const vaultName = await VSCodeUtils.showInputBox({
+        title: "Vault name",
+        prompt: "Name for the new vault",
+        placeHolder: "my-vault",
+      });
+      // If empty, then user cancelled the prompt
+      if (PickerUtilsV2.isInputEmpty(vaultName)) return;
+
+      return {
+        type: sourceType,
+        name: vaultName,
+        path: path.join(
+          FOLDERS.DEPENDENCIES,
+          FOLDERS.LOCAL_DEPENDENCY,
+          vaultName
+        ),
+        isSelfContained: true,
+      };
+    } else {
+      // Remote vault
+      const remote = await VSCodeUtils.showInputBox({
+        title: "Remote URL",
+        prompt: "Enter the URL for the git remote",
+        placeHolder: "git@github.com:dendronhq/dendron.git",
+        ignoreFocusOut: true,
+      });
+      // Cancelled
+      if (PickerUtilsV2.isInputEmpty(remote)) return;
+
+      // Calculate the vault name from the remote. If that fails, ask the user for a unique name to use.
+      const vaultName: string | undefined = GitUtils.getRepoNameFromURL(remote);
+
+      return {
+        type: sourceType,
+        name: vaultName,
+        path: path.join(
+          FOLDERS.DEPENDENCIES,
+          GitUtils.remoteUrlToDependencyPath({
+            vaultName,
+            url: remote,
+          })
+        ),
+        pathRemote: remote,
+        isSelfContained: true,
+      };
+    }
+  }
+
+  async gatherInputs(): Promise<CommandOpts | undefined> {
+    const sourceTypeSelected = await VSCodeUtils.showQuickPick([
+      { label: "local", picked: true },
+      { label: "remote" },
+    ]);
+    if (!sourceTypeSelected) {
+      return;
+    }
+    const sourceType = sourceTypeSelected.label as VaultRemoteSource;
```

Comment:
can we use a type guard to do a check for this instead of a cast? we're moving to not [[Go outside of the typescript compiler|dendron://dendron.docs/dev.process.code.best-practices#going-outside-of-the-typescript-compiler]]

Reply:
Yeah my bad, this issue and the naked strings one. I did not change them since it was existing code, but I'll refactor them to leave the code better than I found it (I think that's a value we have, right? :smile: )

Reply:
pass it forward :)